rust/numaflow-core/src/message.rs

```diff
@@ -1,20 +1,37 @@
-use std::cmp::PartialEq;
-use std::collections::HashMap;
-
+use crate::shared::utils::{prost_timestamp_from_utc, utc_from_timestamp};
+use crate::Error;
+use crate::Result;
 use base64::engine::general_purpose::STANDARD as BASE64_STANDARD;
 use base64::Engine;
+use bytes::Bytes;
 use chrono::{DateTime, Utc};
-
-use crate::shared::utils::{prost_timestamp_from_utc, utc_from_timestamp};
-use crate::Error;
 use numaflow_pb::clients::sink::sink_request::Request;
 use numaflow_pb::clients::sink::Status::{Failure, Fallback, Success};
 use numaflow_pb::clients::sink::{sink_response, SinkRequest, SinkResponse};
 use numaflow_pb::clients::source::{read_response, AckRequest};
 use numaflow_pb::clients::sourcetransformer::SourceTransformRequest;
+use prost::Message as ProtoMessage;
+use serde::{Deserialize, Serialize};
+use std::cmp::PartialEq;
+use std::collections::HashMap;
+use std::sync::OnceLock;
+use std::{env, fmt};
+
+const NUMAFLOW_MONO_VERTEX_NAME: &str = "NUMAFLOW_MONO_VERTEX_NAME";
+const NUMAFLOW_VERTEX_NAME: &str = "NUMAFLOW_VERTEX_NAME";
+
+static VERTEX_NAME: OnceLock<String> = OnceLock::new();
+
+pub(crate) fn get_vertex_name() -> &'static str {
```

Comment:
We have a `mono_vertex_name` in `config.rs` parsed from `metadata.name` . Will this be same ?